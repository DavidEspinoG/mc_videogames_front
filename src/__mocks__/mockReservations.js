export const mockReservation = {
  id: 26,
  user_id: 7,
  videogame_id: 68,
  days: 11,
  total_price: '220.0',
  created_at: '2023-07-20T17:53:13.512Z',
  updated_at: '2023-07-20T17:53:13.512Z',
  videogame: {
    id: 68,
    name: 'Street Fighter',
    photo: 'https://mms.businesswire.com/media/20230122005013/en/1692292/5/p1-1.jpg',
    description:
      'Street Fighter is a 1987 arcade game developed by Capcom. It is the first competitive fighting game produced by the company and the inaugural game in the Street Fighter series.',
    price_per_day: '20.0',
    created_at: '2023-07-20T17:37:47.367Z',
    updated_at: '2023-07-20T17:37:47.367Z',
  },
};

const mockReservations = [
  mockReservation,
  {
    id: 19,
    user_id: 7,
    videogame_id: 62,
    days: 2,
    total_price: '10.0',
    created_at: '2023-07-20T17:37:47.766Z',
    updated_at: '2023-07-20T17:37:47.766Z',
    videogame: {
      id: 62,
      name: 'Tetris',
      photo:
        'https://upload.wikimedia.org/wikipedia/en/thumb/7/7d/Tetris_NES_cover_art.jpg/220px-Tetris_NES_cover_art.jpg',
      description:
        'Tetris is a puzzle video game created in 1984 by Alexey Pajitnov, a Soviet software engineer.',
      price_per_day: '5.0',
      created_at: '2023-07-20T17:37:47.068Z',
      updated_at: '2023-07-20T17:37:47.068Z',
    },
  },
];

export default mockReservations;
