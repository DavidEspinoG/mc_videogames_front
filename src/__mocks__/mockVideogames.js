export const mockVideogame = {
  id: 95,
  name: 'Street Fighter',
  photo: 'https://mms.businesswire.com/media/20230122005013/en/1692292/5/p1-1.jpg',
  description:
    'Street Fighter is a 1987 arcade game developed by Capcom. It is the first competitive fighting game produced by the company and the inaugural game in the Street Fighter series.',
  price_per_day: '20.0',
  created_at: '2023-07-20T21:47:16.294Z',
  updated_at: '2023-07-20T21:47:16.294Z',
};

const mockVideogames = [
  mockVideogame,
  {
    id: 94,
    name: 'Mortal Kombat',
    photo: 'https://thesource.com/wp-content/uploads/2017/10/862771.jpg',
    description:
      'Mortal Kombat is an American media franchise centered on a series of video games, originally developed by Midway Games in 1992.',
    price_per_day: '20.0',
    created_at: '2023-07-20T21:47:16.268Z',
    updated_at: '2023-07-20T21:47:16.268Z',
  },
];

export default mockVideogames;
