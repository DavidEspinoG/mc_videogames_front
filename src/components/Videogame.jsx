import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { deleteVideogame } from '../redux/slices/videogamesSlice';
import '../styles/carousel-item.scss';

const Videogame = ({ data, deleteButton }) => {
  const {
    id, name, photo, description,
  } = data;
  const dispatch = useDispatch();

  const handleDelete = (id) => {
    dispatch(deleteVideogame(id));
  };

  return (
    <article className="col h-100 d-flex flex-column">
      <Link key={id} to={`/details/${id}`}>
        <div className="vh-25 mb-4">
          <img className="h-100 img-fluid rounded aspect-ratio-1" src={photo} alt={name} />
        </div>
        <h3 className="h6 fw-bolder text-uppercase mb-4">{name}</h3>
      </Link>
      <div className="text-muted small">
        <p className="d-webkit-box clamp-3">{description}</p>
      </div>
      {deleteButton && (
        <div className="mt-auto">
          <button type="button" className="btn-delete" onClick={() => handleDelete(id)}>
            Delete
          </button>
        </div>
      )}
    </article>
  );
};

Videogame.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    photo: PropTypes.string,
    description: PropTypes.string,
  }).isRequired,
  deleteButton: PropTypes.bool,
};

Videogame.defaultProps = {
  deleteButton: false,
};

export default Videogame;
