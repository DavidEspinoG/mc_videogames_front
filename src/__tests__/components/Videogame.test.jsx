import Videogame from '../../components/Videogame';
import { mockVideogame } from '../../__mocks__/mockVideogames';
import renderWithProviders from '../test-utils';

describe('Videogame', () => {
  const data = mockVideogame;

  it('should render as expected', () => {
    const { container } = renderWithProviders(<Videogame data={data} />);
    expect(container).toMatchSnapshot();
  });
});
