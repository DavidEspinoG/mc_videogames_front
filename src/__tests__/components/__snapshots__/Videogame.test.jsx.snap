// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Videogame > should render as expected 1`] = `
<div>
  <article
    class="col h-100 d-flex flex-column"
  >
    <a
      href="/details/95"
    >
      <div
        class="vh-25 mb-4"
      >
        <img
          alt="Street Fighter"
          class="h-100 img-fluid rounded aspect-ratio-1"
          src="https://mms.businesswire.com/media/20230122005013/en/1692292/5/p1-1.jpg"
        />
      </div>
      <h3
        class="h6 fw-bolder text-uppercase mb-4"
      >
        Street Fighter
      </h3>
    </a>
    <div
      class="text-muted small"
    >
      <p
        class="d-webkit-box clamp-3"
      >
        Street Fighter is a 1987 arcade game developed by Capcom. It is the first competitive fighting game produced by the company and the inaugural game in the Street Fighter series.
      </p>
    </div>
  </article>
</div>
`;
